### PR TITLE
Upgrade dependencies due to OWASP check

### PR DESCRIPTION
**Pull request by idasbiste**
**Rationale:**
It had been a while since we had upgraded versions in `pkb-common`. In `phr` one of the latest issues raised by the OWASP dependency checker revealed one problem in a transitive library used here. This PR upgrades the dependency checker as well as a lot of dependencies with old versions.
We have also added a couple of suppressions.

### CODE REVIEW

**Review by pkbdevdeploy — COMMENTED** [review]
Version `65-head-6b573e4-337899-SNAPSHOT` was generated from commit `6b573e4b49f409b461e933a6e55907c00136ffed`

**Review by VeryDarkViolet — APPROVED** [review]
LGTM with a few notes where pkb-common is ahead of phr

**Review by jimmymcfox — APPROVED** [review]
Looks good to me, other than the version discrepancies already mentioned by Steve

File: pom.xml [hunk 1]
```diff
@@ -75,14 +75,14 @@
 
         <commons.testing.version>54-8e2e575-247593</commons.testing.version>
         <spring-cloud.version>2021.0.5</spring-cloud.version>
-        <spring.boot.version>2.7.10</spring.boot.version>
+        <spring.boot.version>2.7.14</spring.boot.version>
```
[the remark this answers is not in the document]
Comment (idasbiste) [reply pom.xml:78]: I'll upgrade the other PR.

File: pom.xml [hunk 2]
```diff
@@ -75,14 +75,14 @@
...
         <immutables.version>2.9.0</immutables.version>
-        <checker-qual.version>3.32.0</checker-qual.version>
-        <kotlin.version>1.6.21</kotlin.version>
+        <checker-qual.version>3.37.0</checker-qual.version>
```
[the remark this answers is not in the document]
Comment (idasbiste) [reply pom.xml:80]: I'll upgrade the other PR.

**Pull request merged**